cmd/kubelet/app/server.go

```diff
@@ -121,8 +124,52 @@ is checked every 20 seconds (also configurable with a flag).
 HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 (underspec'd currently) to submit a new manifest.`,
 		Run: func(cmd *cobra.Command, args []string) {
+			// short-circuit on verflag
+			verflag.PrintAndExitIfRequested()
+
+			// TODO(mtaufen): won't need this this once dynamic config is GA
+			// set feature gates so we can check if dynamic config is enabled
+			if err := utilfeature.DefaultFeatureGate.SetFromMap(kubeletServer.KubeletConfiguration.FeatureGates); err != nil {
+				glog.Fatal(err)
+			}
+			// validate the initial KubeletFlags, to make sure the dynamic-config-related flags aren't used unless the feature gate is on
+			if err := options.ValidateKubeletFlags(&kubeletServer.KubeletFlags); err != nil {
+				glog.Fatal(err)
+			}
+			// bootstrap the kubelet config controller, app.BootstrapKubeletConfigController will check
+			// feature gates and only turn on relevant parts of the controller
+			kubeletConfig, kubeletConfigController, err := BootstrapKubeletConfigController(
+				&kubeletServer.KubeletConfiguration, kubeletServer.KubeletFlags.KubeletConfigFile, kubeletServer.KubeletFlags.DynamicConfigDir)
+			if err != nil {
+				glog.Fatal(err)
+			}
+
+			// construct a KubeletServer from kubeletFlags and kubeletConfig
+			kubeletServer.KubeletConfiguration = *kubeletConfig
```

Comment:
I'd prefer we construct the kubeletFlags and kubeletConfig independently, and construct the kubeletServer just before we pass it to Run. The code reads more cleanly if it consists of a sequence of constructions rather than a sequence of mutations to a pre-constructed object.

Reply:
I'd like that too, but I think the order should be 1) split the flag struct out, 2) write code to go from structul to runtime, 3) update command